src/module-1/debounce/index.js

```diff
@@ -1,5 +1,12 @@
 export const debounce = (fn, delay = 0) => {
-
+  let timeout = false;
+  return function() {
+    if (timeout) return;
```

Comment:
Current implementation will ignore subsequent calls of `fn` which is not exactly what we want to reach with `debounce`.  
So, having:  
const fnDebounced = debounce(val => console.info(val), 1000);  
fnDebounced(1);  
fnDebounced(12);  
fnDebounced(123);  
will log `1` only and ignore the rest of the values.  
What we want to achieve here is a call with the latest value we received.  
For example, when user types some search term quickly, we can use debounce to prevent calling search function on each input change. We need to capture the last value, so that a series of  
fnDebounced(1);  
fnDebounced(12);  
fnDebounced(123);  
will call our search func with `123` parameter